src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_archimonde.cpp

```diff
@@ -143,35 +143,37 @@ struct npc_doomfire_spirit : public ScriptedAI
 {
     npc_doomfire_spirit(Creature* creature) : ScriptedAI(creature){ }
 
+    float const m_turnConstant = 0.785402f;
+    float m_fAngle = urand(0, M_PI * 2);
+
     void Reset() override
     {
         scheduler.CancelAll();
         ScheduleTimedEvent(0s, [&] {
-            me->GetMotionMaster()->MovePoint(NEAR_POINT, DoomfireMovement(me->GetPosition()));
-            }, 1500ms);
+            float nextOrientation = Position::NormalizeOrientation(me->GetOrientation() + irand(-1, 1) * m_turnConstant);
+            Position pos = GetFirstRandomAngleCollisionPosition(8.f, nextOrientation); // both orientation and distance verified with sniffs
+            me->NearTeleportTo(pos.GetPositionX(), pos.GetPositionY(), pos.GetPositionZ(), nextOrientation);
+            }, 1600ms);
+
+        m_fAngle = urand(0, M_PI * 2);
     }
 
-    Position DoomfireMovement(Position mePos)
+    Position GetFirstRandomAngleCollisionPosition(float dist, float angle)
     {
-        float angle = mePos.GetOrientation();
-        float distance = 100.0f;
-        float newAngle = angle + ((rand() % 181) - 90) * M_PI / 180;
-        float x = mePos.GetPositionX() + distance * cos(newAngle);
-        float y = mePos.GetPositionY() + distance * sin(newAngle);
-
-        Position targetPos = Position(x, y, me->GetPositionZ());
-        return targetPos;
+        Position pos;
```

Comment:
I would however like to initialize the variable

Reply:
My understanding is that position is a class and not a literal and therefore doesn't need to be initialized, same way we don't initialize ObjectGuid etc.

Reply:
Hmm maybe you are right. Am I wrong for this then https://github.com/azerothcore/azerothcore-wotlk/pull/19244/files#diff-432c5bbfa3061a76766a51f510a0c116a7c996815c479f8615a485aa4b85b336R530 ?

Reply:
Yes, objectguid doesn't need to be initialized